src/overlays/actors/ovl_En_Jsjutan/z_en_jsjutan.c

```diff
@@ -21,15 +27,462 @@ const ActorInit En_Jsjutan_InitVars = {
+
+                if ((i == 0) || isInCreditsScene) {
+                    phi_f2_2 = (3000.0f - distance_2) / 3000.0f;
+                } else {
+                    phi_f2_2 = (2000.0f - distance_2) / 2000.0f;
+                }
+
+                if (phi_f2_2 < 0.0f) {
+                    phi_f2_2 = 0.0f;
+                }
+
+                // temp_f14_2 = (spC8[i] * phi_f2_2) + ((this->unk_170 - (this->unk_170 * phi_f2_2)) - 200.0f);
+                temp_f14_2 = (spC8[i] * phi_f2_2);
+                temp_f14_2 += ((this->unk_170 - (this->unk_170 * phi_f2_2)) - 200.0f);
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Jsjutan/func_80A89860.s")
+                distance_2 = distance_2 - 1500.0f;
+                if (distance_2 < 0.0f) {
+                    distance_2 = 0.0f;
+                }
 
+                phi_f2_3 = 100.0f * distance_2 * 0.01f;
+                if (phi_f2_3 > 100.0f) {
+                    phi_f2_3 = 100.0f;
+                }
+
+                if (temp_f14_2 < phi_f28) {
+                    phi_f28 = temp_f14_2;
+                }
+
+                if (phi_f2_3 < spA8) {
+                    spA8 = phi_f2_3;
+                }
+            }
+        }
+
+        /**
+         * See https://en.wikipedia.org/wiki/Sine_wave#General_form
+         * k: 10000
+         * x: j
+         * w: 4000
+         * t: gameplayFrames
+         * A: spA8
+         * D: phi_f28
+         */
+        temp_f0_3 = Math_SinS(globalCtx->gameplayFrames * 4000 + j * 10000);
+
+        if (this->unk_174) {
+            temp_f14_2 = temp_f0_3 * spA8;
+
+            phi_v1_4 = phi_f28 + (temp_f14_2);
+            temp_a0_3 = (phi_s3->n.ob[1] - this->unk_168) * 50.0f;
+
+            if (phi_v1_4 < temp_a0_3) {
+                phi_v1_4 = temp_a0_3;
+            }
+
+            phi_s0->n.ob[1] = phi_v1_4;
+        } else {
+            temp_f14_2 = temp_f0_3 * spA8;
+
+            phi_s0->n.ob[1] = phi_f28 + (temp_f14_2);
+
+            phi_v1_4 = temp_f14_2 * 0.5f;
+            phi_s0->n.ob[0] = D_80A8EE10[j].x + phi_v1_4;
+            phi_s0->n.ob[2] = D_80A8EE10[j].z + phi_v1_4;
+
+            phi_v1_4 = temp_f14_2;
+            phi_s3->n.ob[0] = D_80A8EE10[j].x + phi_v1_4;
+            phi_s3->n.ob[2] = D_80A8EE10[j].z + phi_v1_4;
+        }
+    }
+
+    // address: ac4 ~ bb8
+    if (!this->unk_174) {
+        this->dyna.actor.velocity.y = 0.0f;
+        this->dyna.actor.world.pos.y = this->unk_168;
+
+        dayTime = gSaveContext.dayTime;
+        if (dayTime >= 0x8000) {
+            dayTime = 0xFFFF - dayTime;
+        }
+
+        this->shadowAlpha = (dayTime * 0.00275f) + 10.0f; // (1.0f / 364.0f) ?
+        this->unk_170 = 1000.0f;
+    } else {
+        Math_ApproachF(&this->dyna.actor.world.pos.y, this->unk_168 - 1000.0f, 1.0f, this->dyna.actor.velocity.y);
+        if (!gSaveContext.dayTime) {
+            if (1) {}
```

Comment:
daytime commonly needs the (void)0 trick. could help with the temp above and some of these other memes. something to keep in mind for the next time you try to match this at least

Reply:
Do I have to come back here?
![image](https://user-images.githubusercontent.com/7416381/111086532-4ee1a300-84fb-11eb-87de-1cedcea3d52e.png)

Reply:
lmfao you can leave it as a note so that whoever tries this later will consider it